game_data_cleaner/game_data_cleaner.py

```diff
@@ -0,0 +1,138 @@
+        games_df = self._clean_best_players(games_df)
+        games_df = self._add_binary_category_flags(games_df)
+        games_df = self._reduce_integers_for_memory_usage(games_df)
+        games_df = self._drop_unreleased(games_df)
+        games_df = self._set_missing_min_players(games_df)
+        games_df, themes_df = self._breakout_themes_df(games_df)
+
+        # Generally we would save the data to a new file, but for now we will just print the info
+        print(games_df.info())
+
+        # We can save the Kaggle dataset to S3 here
+        self.s3_file_handler.save_file(
+            file_path="games/kaggle_data/games.csv", data=games_df
+        )
+
+    def load_games_data(self, file_path: str) -> pd.DataFrame:
+        """Load games data from a file path"""
+        games_df = self.local_handler.load_pkl(file_path)
+        return games_df
+
+    def _drop_duplicates(self, df: pd.DataFrame) -> pd.DataFrame:
+        df = df.copy()
+        df = df.drop_duplicates(subset="BGGId", keep="first")
+        return df
+
+    def _drop_unneeded_columns(self, df: pd.DataFrame) -> pd.DataFrame:
+
+        df = df.copy()
+
+        drop_columns = GAME_CONFIGS["drop_columns"]
+
+        # drop non-boardgame related information
+        for column in drop_columns:
+            if column in df.columns:
+                df = df.drop(column, axis=1)
+
+        return df
+
+    def _clean_best_players(self, df):
+        df = df.copy()
+
+        # fill in missing values with 0
+        df["BestPlayers"] = df["BestPlayers"].fillna("0")
+
+        # if "+" in BestPlayers, get rid of it
+        df["BestPlayers"] = df["BestPlayers"].str.replace("+", "")
+
+        # change the datatype of BestPlayers to int8
+        df["BestPlayers"] = df["BestPlayers"].astype("int8")
+
+        return df
+
+    def _add_binary_category_flags(self, df):
+        df = df.copy()
+
+        for field, category in GAME_CONFIGS["binary_flag_fields"].items():
+            if field in df.columns:
+                df.loc[df[field].notna(), category] = 1
+
+        print(df.head())
+
+        return df
+
+    def _reduce_integers_for_memory_usage(self, df):
+        df = df.copy()
+
+        int_columns = [x for x in GAME_CONFIGS["int_columns"] if x in df.columns]
+        rank_columns = [x for x in GAME_CONFIGS["rank_columns"] if x in df.columns]
+
+        df = integer_reduce(df, int_columns, fill_value=0)
+
+        df = integer_reduce(df, rank_columns, fill_value=df.shape[0] + 1)
+
+        return df
+
+    def _drop_unreleased(self, df: pd.DataFrame) -> pd.DataFrame:
+        df = df.copy()
+
+        # get this year from datetime
+        next_year = datetime.now().year + 1
+
+        df = df[df["YearPublished"] < next_year]
+        return df
+
+    def _set_missing_min_players(self, df: pd.DataFrame) -> pd.DataFrame:
+        df = df.copy()
+        df["MinPlayers"] = df["MinPlayers"].fillna(1)
+        df.loc[df["MinPlayers"] < 1, "MinPlayers"] = 2
```

Comment:
I'm confused by these lines.  The first fills games that have a null value with 1 player.  The second fills games with a less than 1 value with two players.  As long as this is intentional that's fine, but I don't naturally understand why we're treating `Null` and `0` differently.